pallets/parachain-staking/src/lib.rs

```diff
@@ -154,36 +154,110 @@ pub mod pallet {
 		pub top_nominators: Vec<Bond<AccountId, Balance>>,
 		/// Bottom nominators (unbounded), ordered least to greatest
 		pub bottom_nominators: Vec<Bond<AccountId, Balance>>,
-		/// Sum of top nominations + self.bond
+		/// Sum of top delegations + self.bond
 		pub total_counted: Balance,
-		/// Sum of all nominations + self.bond = (total_counted + uncounted)
+		/// Sum of all delegations + self.bond = (total_counted + uncounted)
 		pub total_backing: Balance,
 		/// Current status of the collator
 		pub state: CollatorStatus,
 	}
 
-	/// Convey relevant information describing if a nominator was added to the top or bottom
-	/// Nominations added to the top yield a new total
+	impl<A, B> From<Collator2<A, B>> for CollatorCandidate<A, B> {
+		fn from(other: Collator2<A, B>) -> CollatorCandidate<A, B> {
+			CollatorCandidate {
+				id: other.id,
+				bond: other.bond,
+				delegators: other.nominators,
+				top_delegations: other.top_nominators,
+				bottom_delegations: other.bottom_nominators,
+				total_counted: other.total_counted,
+				total_backing: other.total_backing,
+				request: None,
+				state: other.state,
+			}
+		}
+	}
+
+	#[derive(PartialEq, Clone, Copy, Encode, Decode, RuntimeDebug, TypeInfo)]
+	/// Changes allowed by an active collator candidate to their self bond
+	pub enum CandidateBondChange {
+		Increase,
+		Decrease,
+	}
+
+	#[derive(PartialEq, Clone, Copy, Encode, Decode, RuntimeDebug, TypeInfo)]
+	/// Request scheduled to change the collator candidate self-bond
+	pub struct CandidateBondRequest<Balance> {
+		pub amount: Balance,
+		pub change: CandidateBondChange,
+		pub when: RoundIndex,
+	}
+
+	impl<B> CandidateBondRequest<B> {
+		pub fn new(
+			change: CandidateBondChange,
+			amount: B,
+			when: RoundIndex,
+		) -> CandidateBondRequest<B> {
+			CandidateBondRequest {
+				amount,
+				change,
+				when,
+			}
+		}
+	}
```

Comment:
I think `fn new` is typically useful when the fields aren't `pub`. Here your fields are `pub` so you can create a new instance directly.